datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -287,31 +287,31 @@ pub fn date_trunc(args: &[ColumnarValue]) -> Result<ColumnarValue> {
                 "minute" => {
                     // trunc to minute
                     let second = ScalarValue::TimestampNanosecond(
-                        Some(nano.unwrap() / 1_000_000_000 * 1_000_000_000),
+                        Some(nano.unwrap_or_default() / 1_000_000_000 * 1_000_000_000),
```

Comment:
Currently  in `date_trunc`, `nano` is assumed to be `Some`. To return null for null input instead of `unwrap` on it, we can simply return a null timestamp scalar if `nano` is None.